library/oneview_storage_system_facts.py

```diff
@@ -197,10 +196,8 @@ def execute_module(self):
         else:
             get_method = self.oneview_client.storage_systems.get_by_ip_hostname
 
-        if self.module.params.get('ip_hostname'):
-            storage_systems = get_method(self.module.params['ip_hostname'])
-        elif self.module.params.get('hostname'):
```

Comment:
But these are the OneView's variable names (API200/300 and API500). The user won't be able to copy-paste anymore with this new code.

Reply:
`hostname` appears to be considered as the "default" variable for connecting to an appliance when using REST API modules in `ansible`, or at least I've been told so by the ansible-devel team.

The user should still be able to pass in a `filter` parameter containing `ip_hostname=<something>`, right? As that's the way of the rest api.

In this case we'd just be making it easier by presenting a single variable which can be used to query through the ip in ansible.

Reply:
1. You are connecting to the Storage System 3PAR management appliance, so still it is a hostname, and it is inside the data it won't make any confusion with the OneView hostname.
2.  If the user specifies the name by filter, the Storage System won't be _specific_ anymore, so single resource facts i.e. that run on an instance, won't work.
3. If you still think you should do this change, ensure the storage_system is reflecting it too.